core/src/avm2/array.rs

```diff
@@ -114,6 +114,23 @@ impl<'gc> ArrayStorage<'gc> {
         }
     }
 
+    /// Replace the existing dense storage with a new dense storage
+    /// Does nothing if this `ArrayStorage` is sparse
+    pub fn replace_dense_storage(&mut self, new_storage: Vec<Option<Value<'gc>>>) {
+        let new_occupied_count = new_storage.iter().filter(|v| v.is_some()).count();
+
+        match self {
+            ArrayStorage::Dense {
+                storage,
+                occupied_count,
+            } => {
+                *occupied_count = new_occupied_count;
+                *storage = new_storage;
+            }
+            ArrayStorage::Sparse { .. } => {}
```

Comment:
Can this panic, just to ensure that it doesn't accidentally get called on a Sparse array?

Reply:
Done